pkg/controller/nodenetworkconfigurationpolicy/conditions/manager.go

```diff
@@ -48,20 +57,56 @@ func (m *Manager) NotifySuccess() {
 	}
 }
 
+func (m *Manager) initializeEnactment() (*nmstatev1alpha1.NodeNetworkConfigurationEnactment, error) {
+
+	node := corev1.Node{}
+	err := m.client.Get(context.TODO(), types.NamespacedName{Name: m.nodeName}, &node)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("cannot find node %s", m.nodeName))
+	}
+
+	ownerRefList := []metav1.OwnerReference{
+		{Name: m.policy.Name, Kind: m.policy.Kind, APIVersion: m.policy.APIVersion, UID: m.policy.UID},
+		{Name: m.nodeName, Kind: node.Kind, APIVersion: node.APIVersion, UID: node.UID},
+	}
+
+	enactment := nmstatev1alpha1.NodeNetworkConfigurationEnactment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            m.enactmentName,
+			OwnerReferences: ownerRefList,
+		},
+	}
+
+	err = m.client.Create(context.TODO(), &enactment)
+	if err != nil {
+		return nil, fmt.Errorf("error creating NodeNetworkConfigurationEnactment: %v, %+v", err, enactment)
+	}
+
+	return &enactment, nil
+}
+
 func (m *Manager) updateEnactmentCondition(
-	conditionsSetter func(*nmstatev1alpha1.EnactmentList, string, string),
+	conditionsSetter func(*nmstatev1alpha1.NodeNetworkConfigurationEnactment, string),
 	message string,
 ) error {
 	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		instance := &nmstatev1alpha1.NodeNetworkConfigurationPolicy{}
-		err := m.client.Get(context.TODO(), m.policyName, instance)
+		instance := &nmstatev1alpha1.NodeNetworkConfigurationEnactment{}
+		err := m.client.Get(context.TODO(), types.NamespacedName{Name: m.enactmentName}, instance)
 		if err != nil {
-			return err
+			if !k8s_errors.IsNotFound(err) {
+				return errors.Wrap(err, "enactment read failed")
+			}
+			instance, err = m.initializeEnactment()
```

Comment:
I think we should not initialize a new Enactment from here, that should happen on a higher level.

In this state, the condition manager is responsible of creating the enactment. That would confict with the upper layer where we want to really initialize the enactment with Initial conditions and most important - rendered state. Having initialization on two places makes this confusing.

I suggest we initialize it only on a single place, making sure it exists once at the beginning of the reconciliation (note that it should have all conditions already set to something there). The manager would only update (as the name updateEnactmentCondition suggest), if the enactment it manages does not exist, it would fail.

Reply:
Ack will move enactment creating outside of manager, but at the next PR we will have to do it pre-Reconcile to set the Matching condition.

Reply:
Or just in general when the enactment does not exist (it should when we reconcile a policy that is not being deleted). It is like "Now we initialize our metadata field, this is the initial state".